File: js/18869.js
const fs = require('fs');

const inputs = fs
  .readFileSync('example.txt')
  .toString()
  .trim()
  .split('\n')
  .map((e) => e.split(' ').map(Number));

let idx = 0;
const getLine = () => inputs[idx++];

const [M, N] = getLine();

const arr = [];

for (let i = 0; i < M; i++) {
  const a = getLine();
  const nas = getString([...a]);

  arr.push(nas);
}

let ans = 0;

for (let i = 0; i < M - 1; i++) {
  for (let j = i + 1; j < M; j++) {
    if (arr[i] === arr[j]) ans++;
  }
}

console.log(ans);

function getString(arr) {
  let idx = 1;
  const dic = {};

  const temp = [...arr];

  arr.sort((a, b) => a - b);
  arr.forEach((e) => {
    dic[e] = idx++;
  });

  const result = temp.map((e) => dic[e]);
  return result.join('');
}

File: c++/18869.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

using namespace std;

vector<string> arr;

int main() {
  freopen("example.txt", "r", stdin);
  int M, N;
  cin >> M >> N;

  for (int i = 0; i < M; i++) {
    unordered_map<int, int> dic;
    vector<int> v;
    for (int j = 0; j < N; j++) {
      int a;
      cin >> a;
      v.push_back(a);
    }

    vector<int> temp = v;
    sort(v.begin(), v.end());
    int idx = 1;
    for (int j = 0; j < N; j++) {
      dic[v[j]] = idx++;
    }

    string s;
    for (int j = 0; j < N; j++) {
      s += to_string(dic[temp[j]]);
    }
    arr.push_back(s);
  }

  int ans = 0;
  for (int i = 0; i < M - 1; i++) {
    for (int j = i + 1; j < M; j++) {
      if (arr[i] == arr[j]) ans++;
    }
  }

  cout << ans;

  return 0;
}